compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4058,7 +4058,7 @@ private BType getRemainingType(BTupleType originalType, BTupleType typeToRemove)
             return originalType;
         }
         List<BType> tupleTypes = new ArrayList<>();
-        for (int i = 0; i < originalTupleTypes.size(); i++) {
+        for (int i = 0; i < typesToRemove.size(); i++) {
```

Comment:
Can this be a breaking change? The returned type may change, right?

Reply:
It seems this change is no longer required with the recent changes that have been done for the match patterns. Hence undoing the change.